samples/drivers/ethernet/eth_ivshmem/boards/qemu_cortex_a53.overlay

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023 Enphase Energy
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	soc {
+		/delete-node/ memory@40000000;
+
+		sram0: memory@70000000 {
+			compatible = "mmio-sram";
+			reg = <0x0 0x70000000 0x0 DT_SIZE_M(128)>;
+		};
+
+		/delete-node/ pcie@4010000000;
+
+		pcie: pcie@8e00000 {
+			compatible = "pci-host-ecam-generic";
+			device_type = "pci";
+			reg = <0x00 0x08e00000 0x00 0x100000>;
+			#size-cells = <0x02>;
+			#address-cells = <0x03>;
+			ranges = <0x02000000 0x00 0x10000000 0x0 0x10000000 0x00 0x10000>;
+			#interrupt-cells = <0x01>;
+			interrupt-map-mask = <0x00 0x00 0x00 0x07>;
+			interrupt-map = <
+				0x00 0x00 0x00 1 &gic 0 0 GIC_SPI 108 IRQ_TYPE_EDGE IRQ_DEFAULT_PRIORITY
+				0x00 0x00 0x00 2 &gic 0 0 GIC_SPI 109 IRQ_TYPE_EDGE IRQ_DEFAULT_PRIORITY
+				0x00 0x00 0x00 3 &gic 0 0 GIC_SPI 110 IRQ_TYPE_EDGE IRQ_DEFAULT_PRIORITY
+				0x00 0x00 0x00 4 &gic 0 0 GIC_SPI 111 IRQ_TYPE_EDGE IRQ_DEFAULT_PRIORITY>;
+			bus-range = <0x00 0x00>;
+		};
+	};
+};
+
+&pcie {
+	eth_ivhsmem0: eth_ivhsmem@0 {
```

Comment:
The device tree does not currently have a nice way of dealing with PCI INTx interrupt configuration...

To get this working:

- `eth_ivhsmem0` had to be a child of `pcie`, rather than just `interrupt-parent = <&pcie>;`
- A dummy register address/size was required `reg = <0x00 0x00 0x00 0x00 0x01>;`
- All four INTx interrupts are allocated to allow runtime lookup
- Additional code in the Ethernet driver to lookup the correct interrupt

I think a better solution would be to modify the device tree code to auto-generate the interrupt-map inside the PCI controller driver. This would simplify INTx interrupts usage, reduce duplication and remove the need for these DTS kludges